graphs/graphs_floyd_warshall.py

```diff
@@ -1,62 +1,112 @@
-        distance[k][j]) for each possible pair i, j of vertices.
-    4. The above is repeated for each vertex k in the graph.
-    5. Whenever distance[i][j] is given a new minimum value, next vertex[i][j] is
-        updated to the next vertex[i][k].
+    # doctests:
+
+    >>> graph = [[0, 3, float('inf')], [2, 0, float('inf')],[float('inf'), 7, 0]]
+    >>> floyd_warshall(graph, 3)[0]
+    [[0, 3, inf], [2, 0, inf], [9, 7, 0]]
+
+    >>> graph = [[0, 1, 4],[float('inf'), 0, 2],[float('inf'), float('inf'), 0]]
+    >>> floyd_warshall(graph, 3)[0]
+    [[0, 1, 3], [inf, 0, 2], [inf, inf, 0]]
+
+    >>> graph = [[0, 0, 0],[0, 0, 0],[0, 0, 0]]
+    >>> floyd_warshall(graph, 3)[0]
+    [[0, 0, 0], [0, 0, 0], [0, 0, 0]]
+
+    #Graph with all edge weights = infinity
+
+    >>> graph = [[float('inf'), float('inf'), float('inf')],
+    ... [float('inf'), float('inf'), float('inf')],
+    ... [float('inf'), float('inf'), float('inf')]]
+    >>> floyd_warshall(graph, 3)[0]
+    [[inf, inf, inf], [inf, inf, inf], [inf, inf, inf]]
+
+
+    #Handling negetive weighted graph:
+
+    >>> graph = [[0, -2, float('inf')],[float('inf'), 0, 3],[4, float('inf'), 0]]
+    >>> floyd_warshall(graph, 3)[0]
+    [[0, -2, 1], [7, 0, 3], [4, 2, 0]]
+
+
+    #Handling negetive weighted cycle:
+
+    >>> graph = [[0, -1, float('inf')],[float('inf'), 0, -2],[-3, float('inf'), 0]]
+    >>> floyd_warshall(graph, 3)[0]
+    [[-6, -7, -9], [-5, -6, -8], [-9, -10, -12]]
+
+
+    #Number of vertex in function argument should match number of vertex in graph:
+
+    >>> graph = [[0, -1, float('inf')],[float('inf'), 0, -2]]
+    >>> floyd_warshall(graph, 3)[0]
+    Traceback (most recent call last):
+        ...
+    IndexError: list index out of range
+
+
+    #Graph data type should be a 2D list:
+
+    >>> graph = "strings not allowed"
+    >>> floyd_warshall(graph, 3)[0]
+    Traceback (most recent call last):
+        ...
+    IndexError: string index out of range
     """
 
-    dist = [[float("inf") for _ in range(v)] for _ in range(v)]
+    dist = [[float("inf") for _ in range(vertex)] for _ in range(vertex)]
 
-    for i in range(v):
-        for j in range(v):
+    for i in range(vertex):
+        for j in range(vertex):
             dist[i][j] = graph[i][j]
-
             # check vertex k against all other vertices (i, j)
-    for k in range(v):
+
+    for k in range(vertex):
         # looping through rows of graph array
-        for i in range(v):
+
+        for i in range(vertex):
             # looping through columns of graph array
-            for j in range(v):
+
+            for j in range(vertex):
                 if (
                     dist[i][k] != float("inf")
                     and dist[k][j] != float("inf")
                     and dist[i][k] + dist[k][j] < dist[i][j]
                 ):
                     dist[i][j] = dist[i][k] + dist[k][j]
-
-    _print_dist(dist, v)
-    return dist, v
+    return dist, vertex
```

Comment:
Is `vertex` just the input parameter `vertex`? If this variable isn't being modified in the function, then there's no need to return it again because we already know what the value is.